profile/c5g7-cmfd.cpp

```diff
@@ -0,0 +1,509 @@
+      0.0, 0.330403, 0.0016463, 2.6006E-9, 0.0, 0.0, 0.0,
+      0.0, 0.0, 0.461792, 0.0024749, 0.0, 0.0, 0.0,
+      0.0, 0.0, 0.0, 0.468021, 0.005433, 0.0, 0.0,
+      0.0, 0.0, 0.0, 1.8597E-4, 0.285771, 0.0083973, 8.928E-9,
+      0.0, 0.0, 0.0, 0.0, 0.0023916, 0.247614, 0.012322,
+      0.0, 0.0, 0.0, 0.0, 0.0, 0.0089681, 0.256093}; 
+  chi["MOX-8.7%"] = std::array<double, 7> {0.58791, 0.41176, 3.3906E-4, 
+    1.1761E-7, 0.0, 0.0, 0.0};
+  sigmaT["MOX-8.7%"] = std::array<double, 7> {0.183045, 0.336705, 0.500507, 
+    0.606174, 0.502754, 0.921028, 0.955231};
+
+  /* Define fission chamber cross-sections */
+  sigmaA["Fission Chamber"] = std::array<double, 7> {5.1132E-4, 7.5813E-5, 
+    3.1643E-4, 0.0011675, 0.0033977, 0.0091886, 0.023244};
+  nuSigmaF["Fission Chamber"] = std::array<double, 7> {1.323401E-8, 1.4345E-8, 
+    1.128599E-6, 1.276299E-5, 3.538502E-7, 1.740099E-6, 5.063302E-6};
+  sigmaF["Fission Chamber"] = std::array<double, 7> {4.79002E-9, 5.82564E-9, 
+    4.63719E-7, 5.24406E-6, 1.4539E-7, 7.14972E-7, 2.08041E-6};
+  sigmaS["Fission Chamber"] = std::array<double, 49>
+      {0.0661659, 0.05907, 2.8334E-4, 1.4622E-6, 2.0642E-8, 0.0, 0.0,
+      0.0, 0.240377, 0.052435, 2.499E-4, 1.9239E-5, 2.9875E-6, 4.214E-7,
+      0.0, 0.0, 0.183425, 0.092288, 0.0069365, 0.001079, 2.0543E-4,
+      0.0, 0.0, 0.0, 0.0790769, 0.16999, 0.02586, 0.0049256,
+      0.0, 0.0, 0.0, 3.734E-5, 0.099757, 0.20679, 0.024478,
+      0.0, 0.0, 0.0, 0.0, 9.1742E-4, 0.316774, 0.23876,
+      0.0, 0.0, 0.0, 0.0, 0.0, 0.049793, 1.0991};
+  chi["Fission Chamber"] = std::array<double, 7> {0.58791, 0.41176, 3.3906E-4,
+    1.1761E-7, 0.0, 0.0, 0.0};
+  sigmaT["Fission Chamber"] = std::array<double, 7> {0.126032, 0.29316, 
+    0.28425, 0.28102, 0.33446, 0.56564, 1.17214};
+
+  /* Define guide tube cross-sections */
+  sigmaA["Guide Tube"] = std::array<double, 7> {5.1132E-4, 7.5801E-5, 
+    3.1572E-4, 0.0011582, 0.0033975, 0.0091878, 0.023242};
+  nuSigmaF["Guide Tube"] = std::array<double, 7> {0, 0, 0, 0, 0, 0, 0};
+  sigmaF["Guide Tube"] = std::array<double, 7> {0, 0, 0, 0, 0, 0, 0};
+  sigmaS["Guide Tube"] = std::array<double, 49> 
+      {0.0661659, 0.05907, 2.8334E-4, 1.4622E-6, 2.0642E-8, 0.0, 0.0,
+      0.0, 0.240377, 0.052435, 2.499E-4, 1.9239E-5, 2.9875E-6, 4.214E-7,
+      0.0, 0.0, 0.183297, 0.092397, 0.0069446, 0.0010803, 2.0567E-4,
+      0.0, 0.0, 0.0, 0.0788511, 0.17014, 0.025881, 0.0049297,
+      0.0, 0.0, 0.0, 3.7333E-5, 0.0997372, 0.20679, 0.024478,
+      0.0, 0.0, 0.0, 0.0, 9.1726E-4, 0.316765, 0.23877,
+      0.0, 0.0, 0.0, 0.0, 0.0, 0.049792, 1.09912};
+  chi["Guide Tube"] = std::array<double, 7> {0, 0, 0, 0, 0, 0, 0};
+  sigmaT["Guide Tube"] = std::array<double, 7> {0.126032, 0.29316, 0.28424, 
+    0.28096, 0.33444, 0.56564, 1.17215};
+
+  /* Create materials */
+  log_printf(NORMAL, "Creating materials...");
+  std::map<std::string, Material*> materials;
+  int numEnergyGroups = 7;
```

Comment:
Define this at the top with the other simulation parameters. I think it would also be best to use this variable to define the size of your arrays in a more programmatic (and transparent) fashion, but that isn't necessary here.